protocol-units/execution/opt-executor/src/executor/transaction_pipe.rs

```
@@ -43,68 +39,63 @@ impl Executor {
+						let status = MempoolStatus::new(MempoolStatusCode::MempoolIsFull);
+						callback.send(Ok((status.clone(), None))).map_err(
+							|e| TransactionPipeError::InternalError(format!("Error sending transaction: {:?}", e))
+						)?;
+						return Ok(())
+					}
+
 					let status = {
 						// add to the mempool
-						{
-							let mut core_mempool = self.core_mempool.write().await;
-
-							debug!(
-								"Adding transaction to mempool: {:?} {:?}",
-								transaction,
-								transaction.sequence_number()
-							);
-							let status = core_mempool.add_txn(
-								transaction.clone(),
-								0,
-								transaction.sequence_number(),
-								TimelineState::NonQualified,
-								true,
-							);
-
-							match status.code {
-								MempoolStatusCode::Accepted => {
-									debug!("Transaction accepted: {:?}", transaction);
-								}
-								_ => {
-									debug!("Transaction not accepted: {:?}", status);
-									Err(TransactionPipeError::TransactionNotAccepted(status))?;
-								}
-							}
-						}
-
-						// send along to the receiver
-						transaction_channel
+						let mut core_mempool = self.core_mempool.write().await;
+
+						debug!(
+							"Adding transaction to mempool: {:?} {:?}",
+							transaction,
+							transaction.sequence_number()
+						);
+						core_mempool.add_txn(
+							transaction.clone(),
+							0,
+							transaction.sequence_number(),
+							TimelineState::NonQualified,
+							true,
+						)
+					};
+
+					// increment 
+					match &status.code {
+						MempoolStatusCode::Accepted => {
+							// send along to the receiver
+							transaction_channel
 							.send(transaction)
 							.await
 							.map_err(|e| anyhow::anyhow!("Error sending transaction: {:?}", e))?;
+							self.transactions_in_flight.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+						},
+						_ => {}
```

Silently doing nothing here.

If this case is never supposed to happen, perhaps at least put out a `warn!()` tracing event in case it does?

---

Any comment on this? Is it a legitimate case to silently ignore?